### PR TITLE
feat: .getDepPkgs() to get all pkgs except root

### PR DESCRIPTION
**Pull request by michael-go**
This won't return the root package even if the root is a dependency by itself (a.k.a cycle with root)

### CODE REVIEW

File: test/core/create-from-json.test.ts [hunk 1]
```diff
@@ -26,6 +26,14 @@ describe('fromJSON simple', () => {
       { name: 'e', version: '5.0.0' },
       { name: 'root', version: '0.0.0' },
     ].sort(helpers.depSort));
+    expect(graph.getDepPkgs().sort(helpers.depSort)).toEqual([
```
Comment (robcresswell) [inline test/core/create-from-json.test.ts]: Can you make this a separate test, and change the test name to explain what its actually testing? i.e.

```
describe('getPkgs', () => {
  it('does not return the root node', () => {
    // expect...
  })
})'
```

Also doesn't look like this test needs to be async.

File: src/core/dep-graph.ts [hunk 2]
```diff
@@ -50,6 +51,8 @@ class DepGraphImpl implements types.DepGraphInternal {
     this._rootPkgId = (graph.node(rootNodeId) as Node).pkgId;
 
     this._pkgList = _.values(pkgs);
+    this._depPkgsList = this._pkgList
+      .filter((pkg) => pkg !== this.rootPkg);
```
Comment (robcresswell) [inline src/core/dep-graph.ts]: I don't really know about the internals of a graph, but could you do a `.findIndex` and remove the first occurance of the root package? That would solve the cyclic issue, assuming it appears multiple times?

Comment (michael-go) [reply src/core/dep-graph.ts]: not sure I understand. It won't appear multiple times in this list even if it's cyclic

File: test/core/create-from-json.test.ts (hunk 1 above)
Comment (michael-go) [reply test/core/create-from-json.test.ts]: done. and removed the `async` keyword from all tests in this file

File: src/core/dep-graph.ts (hunk 2 above)
Comment (robcresswell) [reply src/core/dep-graph.ts]: Cool, that makes my suggestion redundant then :)

File: test/core/create-from-json.test.ts (hunk 1 above)
Comment (robcresswell) [reply test/core/create-from-json.test.ts]: 💜

**Pull request merged**